### PR TITLE
Data Handler

### PR DESCRIPTION
**Pull request by Jiarui-Cao**
### Description
Data handlers responsible for writing to and reading from compressed trades/snapshots files.

### Testing
Unit tested with synthetic data and downloaded trade data for AEVOEUR.

### Impact
This should serve as the bottom-level APIs for storing/reading from compressed data files storing historical trades/snapshots data. Raw data loaders (loads trades & snapshots one-by-one) and tick (tick = 1s) data loaders (loads all trades within a tick & last snapshot within the tick) will be built on top of these handlers.

### Other
Should unblock #49 once the data loaders are built on top of this PR.

### CODE REVIEW

File: src/pysrc/test/unit/util/test_historical_data_utils.py
Comment (echavemann) [inline src/pysrc/test/unit/util/test_historical_data_utils.py:1]: i think you can call Path__file__.parent less times

File: src/pysrc/test/unit/data_handlers/kraken/historical/resources/snapshots/XADAZUSD/test.bin
Comment (echavemann) [inline src/pysrc/test/unit/data_handlers/kraken/historical/resources/snapshots/XADAZUSD/test.bin:1]: is there any way we can name them better than test.bin and test2.bin

Comment (Jiarui-Cao) [reply src/pysrc/test/unit/data_handlers/kraken/historical/resources/snapshots/XADAZUSD/test.bin:1]: yes I'll change

File: src/pysrc/test/unit/util/test_historical_data_utils.py
Comment (Jiarui-Cao) [reply src/pysrc/test/unit/util/test_historical_data_utils.py:1]: fair

File: src/pysrc/adapters/messages.py [hunk 1]
```diff
@@ -97,3 +97,28 @@ def __init__(
...
+    @staticmethod
+    def from_bytes(b: bytes, feedcode: str, market: Market) -> "TradeMessage":
+        if len(b) != 17:
+            raise ValueError("Can't create TradeMessage from != 17 bytes")
+        time, price, quantity, side_val = struct.unpack("=QffB", b)
+
+        return TradeMessage(
+            time=time,
+            feedcode=feedcode,
+            n_trades=1,
```
Comment (rewong03) [inline src/pysrc/adapters/messages.py:119]: will assuming n_trades=1 come back to bite us @echavemann

Comment (echavemann) [reply src/pysrc/adapters/messages.py:119]: i dont think so, worst case its less info. if we end up scaling to a point where we need to add an exchange that uses aggtrades thats a W

Comment (Jiarui-Cao) [reply src/pysrc/adapters/messages.py:119]: just curious which exchange aggregates trades and don't tell you how the traded quantities specifically break down?

Comment (echavemann) [reply src/pysrc/adapters/messages.py:121]: they tell u, whether indirectly or directly is a question

**Review by echavemann — APPROVED** [review]

**Pull request merged**